### PR TITLE
Add consul_policy, consul_role and consul_token modules

### LINKED ISSUE #1749: consul_acl in legacy mode only
**Summary**

Hi,

Note: If this is not the correct place to have this discussion, please let me know where to head to.

Tthe `consul_acl` module is still operating in consul acl legacy mode only (or at the very least does not support all new acl possibilities). I went over this limitation by creating my own reusable tasks list using the api but this is quite verbose and not so straightforward to manage idempotency. I was about to start a custom module but I'm wondering it it would be easier to get this into the community collection.

So here are a couple questions to start with:
* Are there any plans on your side to make this module compatible with the recent consul acl system?
* If not, are you still interested in maintaining that module and would you accept contributions in that direction? Or (in other words) are you planning to drop support for this?
* If you intend to keep this module one way or an other, what do you think would be the best way to move on? The new acl system introduces roles, allows you to declare rules in separate reusable policies and finally to declare tokens that will link to policies/roles. Do you think the actual module can be fixed to take all this into account (with possible backward compatibility...)? My actual feeling is that a set of three new modules (e.g. `consul_acl_policy`, `consul_acl_role` and `consul_acl_token`) would be more appropriate.

Looking forward to your comments on this, thanks a million for all these good community ansible utilities.

**Issue Type**

Feature Idea

**Component Name**

consul_acl

**Ansible Version**

```paste below
2.10.5
```

**Configuration**

```paste below

```

**OS / Environment**

_No response_

**Steps To Reproduce**

```paste below

```

**Expected Results**

_No response_

**Actual Results**

```paste below

```

### PR DESCRIPTION
**Pull request by Hakon**
- add consul_policy, consul_role and consul_token modules refactor consul_session to not use the consul python package
- Add some missing documentation, remove python-consul and pyhcl requirements

##### SUMMARY

The consul_acl creates the old type of tokens (consul < 1.4.0) which is deprecated and removed from consul 1.11.0.
The new consul_policy, consul_role and consul_token modules is compatible with the new ACL system. The consul_acl module is kept for backwards compatibility.

The consul_token module _could_ be merged into consul_acl, but it could possibly create backward compatibility issues with existing uses of consul_acl.

I have avoided using any "consul client" python package and integrate with the consul API directly.
This is a tradeoff between maintaining our own client and having 3rd party package dependencies. The consul HTTP is quite stable so I think this is an OK tradeoff to make.

I also refactored the consul_session module to use requests and the HTTP API directly because of a bug where the 3rd party consul client used did not support authentication with tokens. Why this has not been reported before is beyond me, but there is a bug on the python package issue tracker.

Note that I have not implemented a consul_role module but this should be a relatively simple task for someone who wants to get into ansible module development.

Fixes #1749

##### ISSUE TYPE
- New Module Pull Request

##### COMPONENT NAME
consul_policy
consul_role
consul_token

### CODE REVIEW

**Comment by Hakon** [comment]
ready_for_review

File: plugins/modules/clustering/consul/consul_role.py [hunk 1]
```diff
@@ -0,0 +1,476 @@
...
+
+def check_dependencies():
+    """
+    Checks that the required dependencies have been imported.
+    :exception ImportError: if it is detected that any of the required dependencies have not been imported
+    """
+
+    if not has_requests:
+        raise ImportError(
+            "requests required for this module. See https://pypi.org/project/requests/")
```
Comment (felixfontein) [inline plugins/modules/clustering/consul/consul_role.py:426]: The message should be composed by `missing_required_lib()` (`from ansible.module_utils.basic import missing_required_lib`).

File: plugins/modules/clustering/consul/consul_role.py [hunk 2]
```diff
@@ -0,0 +1,476 @@
...
+
+
+def main():
+    """
+    Main method.
+    """
+    module = AnsibleModule(_ARGUMENT_SPEC, supports_check_mode=False)
```
Comment (felixfontein) [inline plugins/modules/clustering/consul/consul_role.py:433]: Generally modules should support check mode if possible. Usually implementing check mode is pretty easy, as you need to do idempotency checks anyway. (The main reason to not support check mode is when idempotency checks are not done by the module, but by the service/software called by the module.)

**Review by felixfontein — COMMENTED** [review]
Thanks for your contribution! I've added some first comments.

Please note that we recommend to not add multiple new modules in one PR, but to instead start with one PR for one new module, get that polished and merged, and then create new PRs for the other modules (and already incorporate all the comments from the first PR). (Generally also see [CONTRIBUTING.md](https://github.com/ansible-collections/community.general/blob/main/CONTRIBUTING.md#creating-new-modules-or-plugins).)

**Comment by Hakon** [comment]
> Please note that we recommend to not add multiple new modules in one PR, but to instead start with one PR for one new module, get that polished and merged, and then create new PRs for the other modules (and already incorporate all the comments from the first PR). (Generally also see [CONTRIBUTING.md](https://github.com/ansible-collections/community.general/blob/main/CONTRIBUTING.md#creating-new-modules-or-plugins).)

Ah, thanks! I thought I had read the CONTRIBUTING.md, but apparently not thoroughly enough. Do you want me to split this into 3 PRs now?

**Comment by felixfontein** [comment]
Please note that in #5461 the collection repository was restructured to remove the directory tree in plugins/modules/, and the corresponding tree in tests/unit/plugins/modules/. Your PR adds new files into this hierarchy. Please **rebase** with the current `main` branch and move your files directly into plugins/modules/. You also can remove the changes to meta/runtime.yml, these are not needed anymore (the corresponding section was removed from CONTRIBUTING.md), and make sure to adjust the new entries in .github/BOTMETA.yml as well.

**Comment by felixfontein** [comment]
Are you still working on this?

needs_info

**Pull request closed without merging**